docs(distribution): New practice to cleanup configuration files (#16)
lirantal/nodejs-cli-apps-best-practices: README.md

</details>

### 2.3 Cleanup configuration files

✅ **Do:**

> Cleanup configuration files when the CLI application gets uninstalled. Optionally CLI application can prompt your users to keep the configuration files to skip the reinitializing phase on the next installation for a better user experience.

Reviewer: ```suggestion
Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.
```

Reviewer: minor wording suggestions

❌ **Otherwise:**

<details>
  <summary>➡️ <b>Details</b></summary>

> As we discussed in the [stateful data section](#13-stateful-data) if your CLI application using storage persistence (configuration files) then the CLI application is responsible for removing its configuration files when it gets uninstalled.

Reviewer: ```suggestion
As mentioned in the [stateful data section](#13-stateful-data), if your CLI application uses persistent storage , such as to save configuration files, then the CLI application should also be responsible for removing its configuration files when it gets uninstalled.
```

- [Do you really know how a lockfile works for yarn and npm packages?](https://snyk.io/blog/making-sense-of-package-lock-files-in-the-npm-ecosystem/)
- [Yarn docs: Should lockfiles be committed to the repository?](https://next.yarnpkg.com/advanced/qa#should-lockfiles-be-committed-to-the-repository)

### 2.3 Cleanup configuration files

✅ **Do:**
Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.

❌ **Otherwise:**

> We want to keep the file system clean of any remaining configuration or identifiable data that the CLI tool might have left in when being installed on the system.

Reviewer: ```suggestion
The user's file system may contain residue in the form of orphaned configuration files, and identifiable data which the CLI tool introduced when installed.
```

Author: I like you suggestion.

ℹ️ **Details**

As mentioned in the [stateful data section](#13-stateful-data), if your CLI application uses persistent storage, such as to save configuration files, then the CLI application should also be responsible for removing its configuration files when it gets uninstalled.

> You can use NPMs `pre` or `post` uninstall [script](https://docs.npmjs.com/misc/scripts) for this purpose. You can find a working example in this [repository](https://github.com/m-sureshraj/jenni/blob/master/src/scripts/pre-uninstall.js).

Reviewer: I like this part, good addition 👌

</details>

### 2.3 Cleanup configuration files

✅ **Do:**
Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.

> ❌ **Otherwise:**

Author: Is that okay to keep this section empty?

Reviewer: @m-sureshraj we should definitely add this. I didn't notice it was empty before.
Can you please update? I would say that we want to keep the file system clean of any remaining configuration or identifiable data that the CLI tool might have left in when being installed on the system.

> ❌ **Otherwise:**

Author: Is that okay to keep this section empty?
